MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.Application.Tests/Api/Controllers/ListControllerApplicationTests.cs

```diff
@@ -0,0 +1,268 @@
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.NoContent));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+
+                _listRepository.Received(1).GetAllItemsAsync();
+                _listRepository.Received(1).DeleteItemAsync(Arg.Is<Guid>(id => id.Equals(itemIdToDelete)));
+                _listRepository.Received(0).GetItemAsync(Arg.Any<Guid>());
+            });
+        }
+
+        [Test]
+        public async Task ReplaceItem_GetAllItems_ReturnsAllItemsIncludedEditedItem()
+        {
+            var id = _items[1].Id;
+            _listRepository.GetAllItemsAsync().Returns(_items);
+            var editedItem = ListItemViewModelBuilder.CreateItem("editedItemText");
+            var updateTime = new DateTime(9528, 12, 28);
+            _timeGenerator.GetCurrentTime().Returns(updateTime, DateTime.MinValue);
+            _guidGenerator.Generate().Returns(id);
+            var replacedItem = _items[1].With(item => item.Text, editedItem.Text).With(item => item.LastUpdateTime, updateTime);
+            var newItemsList = _items.Except(new []{ _items[1] }).Append(replacedItem);
+            _listRepository.ReplaceItemAsync(ArgExtended.IsListItem(replacedItem)).Returns(replacedItem);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.PutAsync(id, editedItem));
+            message1.TryGetContentValue(out IListItem resultItem1);
+            var message2 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message2.TryGetContentValue(out IEnumerable<IListItem> resultItems);
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(resultItem1, Is.EqualTo(replacedItem).UsingListItemComparer());
+                Assert.That(resultItems, Is.EquivalentTo(newItemsList).UsingListItemComparer());
+
+                _listRepository.Received(1).GetAllItemsAsync();
+                _listRepository.Received(1).ReplaceItemAsync(ArgExtended.IsListItem(replacedItem));
+            });
+
+        }
+
+        [Test]
+        public async Task ReplaceNonExistingItem_GetIt_GetAll_WillAddTheItemAndReturnsAllItemsIncludedNewOne()
+        {
+            _listRepository.GetAllItemsAsync().Returns(_items);
+
+            var stringId = "22AC59B7-9517-4EDD-9DDD-EB418A7C1678";
+            var id = Guid.Parse(stringId);
+            var expectedUri = new Uri($"http://www.aaa.com/{stringId}");
+            _location.GetListItemLocation(id).Returns(expectedUri);
+            var newItem = ListItemViewModelBuilder.CreateItem("newItemText");
+            var addedItem = ListItemBuilder.CreateItem(stringId, "newItemText", "2150-12-05");
```

Comment:
You could have reordered these and avoid the `stringId` and `id` completely here…